beacon/manager.go

```diff
@@ -56,11 +56,11 @@ type State interface {
 	HeadSlot() structs.Slot
 	SetHeadSlot(structs.Slot)
 
-	Withdrawals() structs.WithdrawalsState
-	SetWithdrawals(structs.WithdrawalsState)
+	Withdrawals(uint64) structs.WithdrawalsState
+	SetWithdrawals(structs.WithdrawalsState, uint64)
 
-	SetRandao(string)
-	Randao() string
+	SetRandao(string, uint64)
```

Comment:
nitpick - (slot, value) rather than (value, slot) . As slot is more like a key, or context you're passing.